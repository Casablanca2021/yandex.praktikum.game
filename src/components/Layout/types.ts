export type OwnProps = {
    title?: string,
    transparent?: boolean,
}