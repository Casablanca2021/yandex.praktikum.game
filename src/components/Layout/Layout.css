.layout {
  min-height: 100vh;
  display: flex;

  &__app-name {
    font-size: 3em;
    color: #ffffff;
    padding: 4px 8px;
    border-radius: 8px;
    display: inline-block;
    line-height: 1;
    background-color: black;
    font-weight: 700;
    margin-bottom: 12px;
    text-transform: uppercase;
    position: relative;

    &:hover {
      color: #ffffff;
      opacity: 0.8;
    }
  }

  &__app-name-wrapper {
    margin-top: 50px;
    text-align: center;
  }

  &__app-name[data-highlightword]:before {
    content: attr(data-highlightword);
    color: red;
    position: absolute;
    text-shadow: 0 0 5px #000000;
  }

  .ui.layout__container {
    &_valign {
      display: flex;
      flex-direction: column;
      .layout__main {
        flex-grow: 1;
        display: flex;
        align-items: center;
      }
    }
  }
}

.ui.menu.layout__main-menu {
  margin-bottom: 30px;
}

h1.ui.header.layout__header {
  font-size: 2.7em;
}

.main {
  background-color: #ececec;
  border-radius: 4px;
  box-shadow: 0 6px 8px rgba(0, 0, 0, 0.1);
  padding: 15px;

  &_transparent {
    background-color: transparent;
    box-shadow: none;
  }
}
