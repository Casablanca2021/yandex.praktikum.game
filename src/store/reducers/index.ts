import { combineReducers } from 'redux';
import { connectRouter } from 'connected-react-router';

import history from '../history';
import auth from './auth';
import user from './user';
import ssr from './ssr';

const reducer = combineReducers({
  router: connectRouter(history),
  auth,
  user,
  ssr,
});

export default reducer;
