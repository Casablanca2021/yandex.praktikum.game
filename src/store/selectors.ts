import { AppState } from './types';

export const getAuthSelector = (state: AppState) => state.auth;

export const getUserSelector = (state: AppState) => state.user;

export const getSSRSelector = (state: AppState) => state.ssr;
