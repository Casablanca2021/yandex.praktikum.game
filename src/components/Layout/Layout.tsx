import React, { FC } from 'react';
import { Container, Header } from 'semantic-ui-react';
import MainMenu from 'components/MainMenu';
import './Layout.css';
import { t } from 'common/dictionary';
import { Link } from  'react-router-dom';
import { ROUTES } from 'common/consts';
import { OwnProps } from './types';

const Layout: FC<OwnProps> = (props) => {

  const {transparent, children} = props;

  return <div className={`layout ${props.className ?? ''}`}>
    <Container className={props.verticalAlign ? 'layout__container_valign' : ''}>
      <div className="layout__app-name-wrapper">
        <Link to = {ROUTES.ROOT} className="layout__app-name" data-highlightword={t('appName').split(' ')[0]}>{t('appName')}</Link>
      </div>
      <MainMenu />
      {props.title && <Header as='h1' className="layout__header">{props.title}</Header>}
      <div className={`layout__main main ${transparent ? 'main_transparent' : ''}`}>
        {children}
      </div>
    </Container>
  </div>;
};
export default Layout;
