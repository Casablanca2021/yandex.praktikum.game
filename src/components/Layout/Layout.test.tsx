import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from 'store';
import history from 'store/history';
import { ConnectedRouter } from 'connected-react-router';
import Layout from 'components/Layout/Layout';

describe(`Layout Component`, () => {
  it(`renders correctly`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <Layout>
              <div>Hello world</div>
            </Layout>
          </ConnectedRouter>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders correctly with some props`, () => {
    const props = {
      title: 'Title',
    }

    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <Layout {...props}>
              <div>Hello world</div>
            </Layout>
          </ConnectedRouter>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders correctly with title prop`, () => {
    const props = {
      title: 'Title',
    }

    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <Layout {...props}>
              <div>Hello world</div>
            </Layout>
          </ConnectedRouter>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders correctly with all props`, () => {
    const props = {
      title: 'Title',
      transparent: true,
      className: 'class-name',
      verticalAlign: true,
    }

    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <Layout {...props}>
              <div>Hello world</div>
            </Layout>
          </ConnectedRouter>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
