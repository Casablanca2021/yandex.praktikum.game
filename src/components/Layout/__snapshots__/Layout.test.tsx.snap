// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component renders correctly 1`] = `
<div
  className="layout "
>
  <div
    className="ui container"
  >
    <div
      className="layout__app-name-wrapper"
    >
      <a
        className="layout__app-name"
        data-highlightword="Casablanca"
        href="/"
        onClick={[Function]}
      >
        Casablanca racing
      </a>
    </div>
    <div
      className="ui main-menu layout__main-menu menu"
    >
      <a
        className="item main-menu__item"
        onClick={[Function]}
      >
        Главная
      </a>
      <a
        className="item main-menu__item"
        onClick={[Function]}
      >
        Профиль
      </a>
      <a
        className="item main-menu__item"
        onClick={[Function]}
      >
        Форум
      </a>
      <div
        className="right menu"
      >
        <a
          className="item main-menu__item"
          onClick={[Function]}
        >
          Войти
        </a>
        <a
          className="item main-menu__item"
          onClick={[Function]}
        >
          Зарегистрироваться
        </a>
      </div>
    </div>
    <div
      className="layout__main main "
    >
      <div>
        Hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`Layout Component renders correctly with all props 1`] = `
<div
  className="layout class-name"
>
  <div
    className="ui container layout__container_valign"
  >
    <div
      className="layout__app-name-wrapper"
    >
      <a
        className="layout__app-name"
        data-highlightword="Casablanca"
        href="/"
        onClick={[Function]}
      >
        Casablanca racing
      </a>
    </div>
    <div
      className="ui main-menu layout__main-menu menu"
    >
      <a
        className="item main-menu__item"
        onClick={[Function]}
      >
        Главная
      </a>
      <a
        className="item main-menu__item"
        onClick={[Function]}
      >
        Профиль
      </a>
      <a
        className="item main-menu__item"
        onClick={[Function]}
      >
        Форум
      </a>
      <div
        className="right menu"
      >
        <a
          className="item main-menu__item"
          onClick={[Function]}
        >
          Войти
        </a>
        <a
          className="item main-menu__item"
          onClick={[Function]}
        >
          Зарегистрироваться
        </a>
      </div>
    </div>
    <h1
      className="ui header layout__header"
    >
      Title
    </h1>
    <div
      className="layout__main main main_transparent"
    >
      <div>
        Hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`Layout Component renders correctly with some props 1`] = `
<div
  className="layout "
>
  <div
    className="ui container"
  >
    <div
      className="layout__app-name-wrapper"
    >
      <a
        className="layout__app-name"
        data-highlightword="Casablanca"
        href="/"
        onClick={[Function]}
      >
        Casablanca racing
      </a>
    </div>
    <div
      className="ui main-menu layout__main-menu menu"
    >
      <a
        className="item main-menu__item"
        onClick={[Function]}
      >
        Главная
      </a>
      <a
        className="item main-menu__item"
        onClick={[Function]}
      >
        Профиль
      </a>
      <a
        className="item main-menu__item"
        onClick={[Function]}
      >
        Форум
      </a>
      <div
        className="right menu"
      >
        <a
          className="item main-menu__item"
          onClick={[Function]}
        >
          Войти
        </a>
        <a
          className="item main-menu__item"
          onClick={[Function]}
        >
          Зарегистрироваться
        </a>
      </div>
    </div>
    <h1
      className="ui header layout__header"
    >
      Title
    </h1>
    <div
      className="layout__main main "
    >
      <div>
        Hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`Layout Component renders correctly with title prop 1`] = `
<div
  className="layout "
>
  <div
    className="ui container"
  >
    <div
      className="layout__app-name-wrapper"
    >
      <a
        className="layout__app-name"
        data-highlightword="Casablanca"
        href="/"
        onClick={[Function]}
      >
        Casablanca racing
      </a>
    </div>
    <div
      className="ui main-menu layout__main-menu menu"
    >
      <a
        className="item main-menu__item"
        onClick={[Function]}
      >
        Главная
      </a>
      <a
        className="item main-menu__item"
        onClick={[Function]}
      >
        Профиль
      </a>
      <a
        className="item main-menu__item"
        onClick={[Function]}
      >
        Форум
      </a>
      <div
        className="right menu"
      >
        <a
          className="item main-menu__item"
          onClick={[Function]}
        >
          Войти
        </a>
        <a
          className="item main-menu__item"
          onClick={[Function]}
        >
          Зарегистрироваться
        </a>
      </div>
    </div>
    <h1
      className="ui header layout__header"
    >
      Title
    </h1>
    <div
      className="layout__main main "
    >
      <div>
        Hello world
      </div>
    </div>
  </div>
</div>
`;
